### PR TITLE
feat: support multiple project setup projects

### PR DESCRIPTION
**Pull request by Jefiozie**
Will update the description after our call

### CODE REVIEW

File: schematics/utils.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { virtualFs, workspaces } from '@angular-devkit/core';
+import { Tree } from '@angular-devkit/schematics';
+
+// this code is orginal from @schematics/angular
+// we also could include the lib and provide the func getWorkspace from their
```
Comment (bampakoa) [inline schematics/utils.ts:5]: I would definitely recommend to use it directly from `@schematics/angular` 💯

**Comment by bampakoa** [comment]
@Jefiozie this is what I had in mind from your contribution at the first place 🥰 . You see, I am left a bit behind with schematics and I used some old techniques for getting the workspace, project and so on. But I can see from your changes that the API has really changed and became much simpler. Thanks so much for this 🙏

File: package.json [hunk 2]
```diff
@@ -12,6 +12,9 @@
   "homepage": "https://github.com/bampakoa/ngx-electronify#readme",
   "main": "dist/index.js",
   "builders": "builders.json",
+  "ng-add": {
+    "save": "dependencies"
```
Comment (bampakoa) [inline package.json:16]: Does this mean that essentially that `ng add` will also install it as a dependency?

Comment (Jefiozie) [reply package.json:16]: Yes it does, let me show you the part in the `angular-cli` that does this.

File: schematics/utils.ts (hunk 1 above)
Comment (Jefiozie) [reply schematics/utils.ts:5]: Let's discuss this in our session 😀

**Comment by bampakoa** [comment]
@Jefiozie could you please rebase that too? 🙏

**Comment by Jefiozie** [comment]
I've rebased it and I've also added the `getWorkspace` function again. In the `@schematics/angular` package there is no host returned, this is needed to update the `angular.json` file. Maybe we can refactor this in a later stadium?

File: schematics/index.ts [hunk 3]
```diff
@@ -1,45 +1,41 @@
+import { workspaces } from '@angular-devkit/core';
 import {
-  chain,
-  Rule,
   SchematicContext,
   SchematicsException,
   Tree
 } from '@angular-devkit/schematics';
-import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import { Schema } from "./schema";
```
Comment (bampakoa) [inline schematics/index.ts:7]:
```suggestion
import { Schema } from './schema';
```

File: schematics/index.ts [hunk 4]
```diff
@@ -1,45 +1,41 @@
...
+    );
+  }
 
-function getWorkspace(host: Tree) {
-  const workspaceConfig = host.read('/angular.json');
-  if (!workspaceConfig) {
-    throw new SchematicsException();
+  if (project.extensions.projectType !== 'application') {
+    throw new SchematicsException(
+      `Electron requires an Angular project type of "application" in angular.json.`
```
Comment (bampakoa) [inline schematics/index.ts:29]:
```suggestion
      `ngx-electronify requires an Angular project type of "application" in angular.json.`
```

File: schematics/utility.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import { virtualFs, workspaces } from "@angular-devkit/core";
+import { Tree } from "@angular-devkit/schematics";
```
Comment (bampakoa) [inline schematics/utility.ts:2]:
```suggestion
import { virtualFs, workspaces } from '@angular-devkit/core';
import { Tree } from '@angular-devkit/schematics';
```

**Review by bampakoa — CHANGES_REQUESTED** [review]
Sure @Jefiozie 👍 I have added some suggestions and we are good to go 🚀

**Comment by Jefiozie** [comment]
We're all set now!

**Review by bampakoa — APPROVED** [review]
@Jefiozie I took the initiative and merged master to the PR because `package-lock.json` had conflicts again.

**Comment by Jefiozie** [comment]
Thank you, let me know if I need to change something

**Pull request merged**